### PR TITLE
[#2153] improvement: enable DefaultCharset error-prone check

### LINKED ISSUE #2153: [Improvement] in HTTPClient.java character set should be specified
### What would you like to be improved?

HTTPClient.java:336: warning: [DefaultCharset] Implicit use of the platform default charset, which can result in differing behaviour between JVM executions or incorrect behavior if the encoding of the data source doesn't match expectations.
```java
AuthConstants.HTTP_HEADER_AUTHORIZATION, new String(authDataProvider.getTokenData()));
```

 Did you mean 'AuthConstants.HTTP_HEADER_AUTHORIZATION, new String(authDataProvider.getTokenData(), UTF_8));' or 'AuthConstants.HTTP_HEADER_AUTHORIZATION, new String(authDataProvider.getTokenData(), Charset.defaultCharset()));'?

See https://errorprone.info/bugpattern/DefaultCharset

### How should we improve?

Select one to use

### PR DESCRIPTION
**Pull request by coolderli**
### What changes were proposed in this pull request?

- close #2153 

### Why are the changes needed?

Fix: #2153 

### Does this PR introduce _any_ user-facing change?

- no

### How was this patch tested?

- `./gradlew build`

### CODE REVIEW

**Comment by coolderli** [comment]
@justinmclean Please take a look when you have time. Thanks.

**Review by qqqttt123 — APPROVED** [review]
LGTM.

File: catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/miniHMS/ScriptRunner.java [hunk 1]
```diff
@@ -40,7 +40,10 @@ public class ScriptRunner {
   private final boolean stopOnError;
   private final boolean autoCommit;
 
+  @SuppressWarnings("DefaultCharset")
```
Comment (qqqttt123) [inline catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/miniHMS/ScriptRunner.java:43]: Why do we ignore this?

Comment (coolderli) [reply catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/miniHMS/ScriptRunner.java:43]: @qqqttt123 Of course we can fix it. But I'm not sure it's necessary. Here is the suggestion:
```
/home/mi/IdeaProjects/gravitino/catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/miniHMS/ScriptRunner.java:43: 警告: [DefaultCharset] Implicit use of the plaorm default charset, which can result in differing behaviour between JVM executions or incorrect behavior if the encoding of the data source doesn't match expectations.
  private final PrintWriter logWriter = new PrintWriter(System.out);
                                        ^
    (see https://errorprone.info/bugpattern/DefaultCharset)
  Did you mean 'private final PrintWriter logWriter = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out, UTF_8)));' or 'private final PrintWriter logWriter = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out, Charset.defaultCharset())));'?

```

What do you think? Should I fix it?

Comment (qqqttt123) [reply catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/miniHMS/ScriptRunner.java:43]: We would better fix it.

Comment (coolderli) [reply catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/miniHMS/ScriptRunner.java]: Done. I have fixed it.

File: core/src/test/java/com/datastrato/gravitino/storage/kv/TestTransactionalKvBackend.java [hunk 2]
```diff
@@ -39,6 +39,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+@SuppressWarnings("DefaultCharset")
```
Comment (jerryshao) [inline core/src/test/java/com/datastrato/gravitino/storage/kv/TestTransactionalKvBackend.java:42]: Also here.

File: core/src/test/java/com/datastrato/gravitino/storage/kv/TestKvGarbageCollector.java [hunk 3]
```diff
@@ -42,6 +43,7 @@
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
+@SuppressWarnings("DefaultCharset")
```
[the remark this answers is not in the document]
Comment (coolderli) [reply core/src/test/java/com/datastrato/gravitino/storage/kv/TestKvGarbageCollector.java:46]: This file will take a considerable amount of time to repair. Considering it is a test file, I ignore it. Set `@SuppressWarnings("DefaultCharset")` on the class , this file will skip the error-prone check.

[the remark this answers is not in the document]
Comment (jerryshao) [reply core/src/test/java/com/datastrato/gravitino/storage/kv/TestKvGarbageCollector.java:46]: I see.

**Review by jerryshao — APPROVED** [review]
LGTM, thanks for your contribution @coolderli

**Pull request merged**